### ISSUE TITLE
Judder / stutter during dolly shot

### ISSUE THREAD

**Issue by cbs228** [daipenger/minema#33]
I am experiencing a recording issue with the following combination of mods:

* Minema-3.5.4-1.12.2 (also tested 3.6.1)
* aperture-1.3.6-1.12.2 (also tested 1.6.1)
* OptiFine_1.12.2_HD_U_G5
* forge-1.12.2-14.23.5.2855 on openjdk 1.8.0_302 on various versions of Ubuntu

I am recording on a singleplayer world with sync enabled.

When recording a pre-programmed dolly shot with Aperture, there is a timing glitch in the recorded video that recurs about once per second. The camera appears to jump or skip frames. 
Here is an [example glitch](https://user-images.githubusercontent.com/3526918/144732332-08a16602-4a57-4649-a821-394467008fa9.mp4) at 60 fps and another take [with F3](https://user-images.githubusercontent.com/3526918/144732335-d83bf52d-e2f4-40fc-abf2-85cf4f63a314.mp4) enabled.

This could very well be an aperture issue, but the periodicity suggests it might be something that happens during capture. I cannot find any gap in the frame numbers in the recording, however.

I don't know if Minema is causing the problem, but I know what *isn't* causing the problem:

- **ffmpeg**: tried with and without video encoding. Videos encoded from raw images is also glitched
- **shaders**: I have tried with and without OptiFine shaders. I have the anti-aliasing fix turned on, but I have also tried without it
- **gpu**: I have tried NVIDIA hardware, Mesa Intel UHD, and Mesa llvmpipe

The issue is more noticeable at 60 fps but is still visible in [30 fps recordings](https://user-images.githubusercontent.com/3526918/144732821-e36d951b-0327-471e-9dc0-22713d131eb7.mp4).

One thing I have observed is that the `pU` ("pending upload") count in F3 occasionally becomes nonnegative. Can these pending upload frames be lost before they are captured?

I suppose this could be a display issue specific to my monitor, which runs at either 60.03 Hz or 59.96 Hz. Using a fractional framerate doesn't really seem to help.

Any advice would be helpful.

**Comment by mchorse** [comment]
I appreciate the amount of effort and time you put into the report, however unfortunately I didn't find any issues with the video. 

I didn't check fully but it seems like every frame has exactly 0.029 (or ~0.03) X step, judging from the F3 video provided, and the Frame label has the same exact amount of frames as the length of the video (I checked in Adobe Premiere, last visible frame was at 8:25 which is `505` frames, and last frame shown `Frame: 504` which is exactly `505` frames if you count also `Frame: 0`).

I don't think it's a Minema issue as on my side I don't have any frame skips neither in video editor nor in the media player (sometimes only the first time due to preloading data). 

My wild guess would be it's either due to:

1. The monitor's refresh rate
2. Some Linux stuff
3. High bitrate of the video

I have the most certainty it's `3.`, Minema records at very high bitrates by defaukt, maybe the amount of data is causing those stutters? I can't judge for sure as I don't know whether GitHub compressed these videos or if you did it manually. Hopefully that helps.

**Comment by cbs228** [comment]
@mchorse, thanks for getting back to me so quickly. I configured minema to shoot lossless x264 with `-crf 0`. The source videos do indeed have very high data rate. Before uploading the videos, I recompressed them with

```bash
ffmpeg -i INFILE.mp4 -vcodec libx264 -preset medium -qp 18 OUTFILE.mp4
```

And I still see the issue on the recompressed video. I have also observed the issue when encoding a raw image series directly with `-qp 18`. I see it with VP9 recompression, too:

```bash
ffmpeg -i INFILE.mp4 -vcodec=libvpx-vp9 -crf 27 \
   -vb 1900k -minrate 1300k -maxrate 3000k -g 600 \
  OUTFILE.webm
```

Perhaps the data rate is still too high. Maybe there's an issue recompressing the lossless video. I have found that the quality metrics don't seem to work well for Minecraft videos. (Probably because they are optimized for "live-action" cinema.)

This [judder test video](https://www.youtube.com/watch?v=hkN3PRPh00E) on youtube plays mostly smooth for me, and it has a much higher average data rate than my videos. I have plenty of CPU and I/O margin for video. The youtube framerate is slightly lower: 60000/1001 instead of 60.0. This might matter.

Since my video plays smoothly for you, I'm going to close this issue as invalid. I suspect that this is either a pulldown/refresh rate issue with my hardware or some problem with the encoding. If you have any encoding tips, I'd certainly appreciate them.
